src/openforms/forms/tests/e2e_tests/test_form_designer.py

```diff
@@ -351,6 +351,48 @@ async def test_enable_translations_and_create_new_step(self):
             # the modal should close
             await expect(page.locator("css=.formio-dialog-content")).to_be_hidden()
 
+    async def test_key_automatically_generated_for_select_options(self):
+        @sync_to_async
+        def setUpTestData():
+            # set up a form
+            form = FormFactory.create(
+                name="Playwright test",
+                name_nl="Playwright test",
+                generate_minimal_setup=True,
+                formstep__form_definition__name_nl="Playwright test",
+                formstep__form_definition__configuration={
+                    "components": [],
+                },
+            )
+            return form
+
+        await create_superuser()
+        form = await setUpTestData()
+        admin_url = str(
+            furl(self.live_server_url)
+            / reverse("admin:forms_form_change", args=(form.pk,))
+        )
+
+        async with browser_page() as page:
+            await self._admin_login(page)
+            await page.goto(str(admin_url))
+            await page.get_by_role("tab", name="Steps and fields").click()
+
+            # Drag and drop a component
+            await drag_and_drop_component(page, "Keuzelijst")
+
+            # Check that the modal is open
+            await expect(page.locator("css=.formio-dialog-content")).to_have_count(1)
```

Comment:
I found out about `.to_be_visible()` which is semantically closer than `to_have_count`, but I'll clean that up in a separate PR for all the existing tests I think :thinking: